packages/flutter_tools/lib/src/application_package.dart

```diff
@@ -349,34 +360,128 @@ class ApplicationPackageStore {
   }
 }
 
+class _Entry {
+  _Element parent;
+  int level;
+}
+
+class _Element extends _Entry {
+  List<_Entry> children;
+  String name;
+
+  _Element.fromLine(String line, _Element parent) {
+    //      E: application (line=29)
+    final List<String> parts = line.trimLeft().split(' ');
+    name = parts[1];
+    level = line.length - line.trimLeft().length;
+    this.parent = parent;
+    children = <_Entry>[];
+  }
+
+  void addChild(_Entry child) {
+    children.add(child);
+  }
+
+  _Attribute firstAttribute(String name) {
+    return children.firstWhere(
+        (_Entry e) => e is _Attribute && e.key.startsWith(name),
+        orElse: () => null);
+  }
+
+  _Element firstElement(String name) {
+    return children.firstWhere(
+        (_Entry e) => e is _Element && e.name.startsWith(name),
+        orElse: () => null);
+  }
+
+  Iterable<_Entry> allElements(String name) {
+    return children.where(
+            (_Entry e) => e is _Element && e.name.startsWith(name));
+  }
+}
+
+class _Attribute extends _Entry {
+  String key;
+  String value;
+
+  _Attribute.fromLine(String line, _Element parent) {
+    //     A: android:label(0x01010001)="hello_world" (Raw: "hello_world")
+    final List<String> keyVal = line.substring(line.indexOf('A: ') + 3).split('=');
```

Comment:
3 is a magic constant---I suppose it is a shorthand for `'A: '.length`?
Maybe extract `'A: '` as a `const String attributePrefix = 'A: ';` and use that in both places.